### PR TITLE
SDK-1611: Provide information about skipped manifests

### PR DESCRIPTION
**Pull request by VladislavStrelchenko**
Ticket: https://spryker.atlassian.net/browse/SDK-1611

Change log: 
* Added information about skipped manifests to an PR description

### CODE REVIEW

File: src/Upgrade/Application/Dto/IntegratorResponseDto.php [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        return $this->data[static::MESSAGES_KEY];
+    }
+
+    /**
+     * @return array<string>|null
+     */
+    public function getWarnings(): ?array
+    {
+        if (!is_array($this->data[static::WARNINGS_KEY])) {
+            var_dump($this->data);
```
Comment (vol4onok) [inline src/Upgrade/Application/Dto/IntegratorResponseDto.php:55]: remove it

File: src/Upgrade/Application/Dto/IntegratorResponseDto.php [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+/**
+ * Copyright © 2016-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+declare(strict_types=1);
+
+namespace Upgrade\Application\Dto;
+
+class IntegratorResponseDto
```
Comment (vol4onok) [inline src/Upgrade/Application/Dto/IntegratorResponseDto.php:12]: strange use of data transfer. if you don't want to use output you can create an adapter.

Comment (vol4onok) [reply src/Upgrade/Application/Dto/IntegratorResponseDto.php:12]: or use set-get for properties

File: src/Upgrade/Infrastructure/Adapter/IntegratorAdapter.php [hunk 3]
```diff
@@ -11,6 +11,7 @@
 
 use Core\Infrastructure\Service\ProcessRunnerServiceInterface;
 use Upgrade\Application\Adapter\IntegratorAdapterInterface;
+use Upgrade\Application\Dto\IntegratorResponseDto;
 use Upgrade\Application\Dto\StepsResponseDto;
 
 class IntegratorAdapter implements IntegratorAdapterInterface
```
Comment (vol4onok) [inline src/Upgrade/Infrastructure/Adapter/IntegratorAdapter.php:17]: btw,  it not an adapter :) seems like we already use that

File: src/Upgrade/Application/Dto/IntegratorResponseDto.php (hunk 1 above)
Comment (VladislavStrelchenko) [reply src/Upgrade/Application/Dto/IntegratorResponseDto.php:55]: Thanks ++

File: src/Upgrade/Application/Dto/IntegratorResponseDto.php (hunk 2 above)
Comment (VladislavStrelchenko) [reply src/Upgrade/Application/Dto/IntegratorResponseDto.php:12]: You are right, usually, we use some builder for DTO 
In this case, the response structure is so simple, and it's not necessary and I want to follow the same structure as in ComposerLockDiffDto

File: src/Upgrade/Infrastructure/Adapter/IntegratorAdapter.php (hunk 3 above)
Comment (VladislavStrelchenko) [reply src/Upgrade/Infrastructure/Adapter/IntegratorAdapter.php:17]: renamed to IntegratorExecutor

File: src/Upgrade/Infrastructure/VersionControlSystem/Generator/PullRequestDataGenerator.php [hunk 4]
```diff
@@ -10,23 +10,41 @@
...
             . PHP_EOL
             . '#### Overview'
             . PHP_EOL;
 
+        if ($majorAvailableInfo) {
+            $text .= '​' . PHP_EOL . '**Available majors:**' . PHP_EOL . PHP_EOL;
+            $text .= $majorAvailableInfo;
+        }
+
+        if ($integratorResponseDto && $integratorResponseDto->getWarnings()) {
```
Comment (pavelmaksimov25) [inline src/Upgrade/Infrastructure/VersionControlSystem/Generator/PullRequestDataGenerator.php:40]: why don't we use twig instead?

File: src/Upgrade/Application/Dto/IntegratorResponseDto.php [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+     */
+    public function __construct(array $data = [])
+    {
+        $this->data = $data;
+    }
+
+    /**
+     * @return array<string>|null
+     */
+    public function getMessages(): ?array
```
Comment (pavelmaksimov25) [inline src/Upgrade/Application/Dto/IntegratorResponseDto.php:40]: `get*` method name, usually mean return smth or fail.
for logic, that is used here, `findMessages()` more suitable.

The same for the `getWarnings()`

File: src/Upgrade/Application/Dto/IntegratorResponseDto.php [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    {
+        if (!is_array($this->data[static::MESSAGES_KEY])) {
+            return null;
```
Comment (pavelmaksimov25) [inline src/Upgrade/Application/Dto/IntegratorResponseDto.php:43]: actually, it is not necessary to return null here. You can return an empty array instead and have a single return type.

File: src/Upgrade/Application/Dto/StepsResponseDto.php [hunk 7]
```diff
@@ -26,10 +26,15 @@ class StepsResponseDto extends ResponseDto
...
+    /**
+     * @var \Upgrade\Application\Dto\IntegratorResponseDto|null
+     */
+    protected ?IntegratorResponseDto $integratorResponseDto = null;
+
     /**
      * @var string|null
      */
-    protected ?string $pullRequestWarning = null;
+    protected ?string $majorAvailableInfo = null;
```
Comment (pavelmaksimov25) [inline src/Upgrade/Application/Dto/StepsResponseDto.php:37]: I'd suggest using an empty string as a default value and making it `string` only will simplify the code and make it cleaner.

File: src/Upgrade/Infrastructure/VersionControlSystem/Generator/PullRequestDataGenerator.php (hunk 4 above)
Comment (VladislavStrelchenko) [reply src/Upgrade/Infrastructure/VersionControlSystem/Generator/PullRequestDataGenerator.php:40]: I want to add only one section here
don't want to rewrite all section

File: src/Upgrade/Application/Dto/IntegratorResponseDto.php (hunk 5 above)
Comment (VladislavStrelchenko) [reply src/Upgrade/Application/Dto/IntegratorResponseDto.php:40]: For the purpose, we use the naming `getOrFail`

File: src/Upgrade/Application/Dto/IntegratorResponseDto.php (hunk 1 above)
Comment (Kyrylo-Meretskyi) [inline src/Upgrade/Application/Dto/IntegratorResponseDto.php:55]: Seems you forgot to remove `var_dump`

File: src/Upgrade/Application/Dto/StepsResponseDto.php [hunk 8]
```diff
@@ -114,6 +119,24 @@ public function getComposerLockDiff(): ?ComposerLockDiffDto
...
+    }
+
+    /**
+     * @param \Upgrade\Application\Dto\IntegratorResponseDto|null $integratorResponseDto
+     *
+     * @return void
+     */
+    public function setIntegratorResponseDto(?IntegratorResponseDto $integratorResponseDto): void
+    {
+        $this->integratorResponseDto = $integratorResponseDto;
```
Comment (Kyrylo-Meretskyi) [inline src/Upgrade/Application/Dto/StepsResponseDto.php:137]: add `return $this` to have consistence setters in this class

File: src/Upgrade/Application/Dto/IntegratorResponseDto.php (hunk 6 above)
Comment (VladislavStrelchenko) [reply src/Upgrade/Application/Dto/IntegratorResponseDto.php:43]: Thanks ++

File: src/Upgrade/Application/Dto/StepsResponseDto.php (hunk 7 above)
Comment (VladislavStrelchenko) [reply src/Upgrade/Application/Dto/StepsResponseDto.php]: ++

File: src/Upgrade/Application/Dto/StepsResponseDto.php (hunk 8 above)
Comment (VladislavStrelchenko) [reply src/Upgrade/Application/Dto/StepsResponseDto.php:137]: Thanks ++

**Review by Kyrylo-Meretskyi — APPROVED** [review]

**Review by pavelmaksimov25 — APPROVED** [review]

**Review by vol4onok — APPROVED** [review]

**Pull request merged**